addons/delivery/models/delivery_carrier.py

```diff
@@ -226,8 +226,9 @@ def rate_shipment(self, order):
             res['carrier_price'] = res['price']
             # free when order is large enough
             amount_without_delivery = order._compute_amount_total_without_delivery()
-            if res['success'] and self.free_over and self._compute_currency(order, amount_without_delivery, 'pricelist_to_company') >= self.amount:
-                res['warning_message'] = _('The shipping is free since the order amount exceeds %.2f.') % (self.amount)
+            if (res['success'] and (self.free_over and self.delivery_type != 'base_on_rule')
+                  and self._compute_currency(order, amount_without_delivery, 'pricelist_to_company') >= self.amount):
```

Comment:
```suggestion
            if (
                res['success']
                and self.free_over
                and self.delivery_type != 'base_on_rule'
                and self._compute_currency(order, amount_without_delivery, 'pricelist_to_company') >= self.amount
            ):
```
It seems like some parenthesis are not necessary (and the whole condition would be clearer to understand if indented).